app/models/operation.rb

```diff
@@ -1,3 +1,25 @@
 class Operation < ApplicationRecord
   belongs_to :tank
+  belongs_to :family, required: false
+
+  validates :family, presence: true, if: :add_family?
```

Comment:
@jtu0 / @mdworken - Thoughts on this being `if: :needs_family?` to more clearly express intent?